From aee7422d5188c08208c12b700b7dc54109beaaa1 Mon Sep 17 00:00:00 2001
From: "Mishra, ManishX" <manishx.mishra@intel.com>
Date: Thu, 25 Nov 2021 13:31:35 +0530
Subject: [PATCH] Hvec Decoder fix for EHL

This is a WA solution for one GPU hang issue for EHL
android.security.cts.StagefrightTest#testBug_63045918.

Tracked-On: OAM-100159
Signed-off-by: Mishra, ManishX <manishx.mishra@intel.com>
---
 omx_components/src/mfx_omx_vdec_component.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/omx_components/src/mfx_omx_vdec_component.cpp b/omx_components/src/mfx_omx_vdec_component.cpp
index e4872eb..6a4a6a3 100644
--- a/omx_components/src/mfx_omx_vdec_component.cpp
+++ b/omx_components/src/mfx_omx_vdec_component.cpp
@@ -2217,7 +2217,7 @@ mfxStatus MfxOmxVdecComponent::InitCodec(void)
             }
         }
         // A WA for one GPU hang issue for android.security.cts.StagefrightTest#testBug_63045918
-        if (m_bUseSystemMemory && (MFX_HW_TGL_LP == m_pDevice->GetPlatformType()) && (MFX_CODEC_HEVC == m_MfxVideoParams.mfx.CodecId)
+        if (m_bUseSystemMemory && (MFX_HW_TGL_LP == m_pDevice->GetPlatformType() || MFX_HW_EHL == m_pDevice->GetPlatformType()) && (MFX_CODEC_HEVC == m_MfxVideoParams.mfx.CodecId)
                 && (MFX_PROFILE_HEVC_MAIN == m_MfxVideoParams.mfx.CodecProfile) && (MFX_LEVEL_HEVC_62 == m_MfxVideoParams.mfx.CodecLevel)
                 && (416 == m_MfxVideoParams.mfx.FrameInfo.Width) && (240 == m_MfxVideoParams.mfx.FrameInfo.Height))
         {
-- 
2.34.0

